tests/test_core.py

```diff
@@ -72,6 +72,20 @@ def decorated(self):
   return decorated
 
 
+# without EMTEST_ALL_ENGINES set we only run tests in a single VM by
+# default. in some tests we know that cross-VM differences may happen and
+# so are worth testing, and they should be marked with this decorator
+def all_engines(f):
+  def decorated(self):
+    old = self.use_all_engines
+    self.use_all_engines = True
+    try:
+      f(self)
+    finally:
+      self.use_all_engines = old
```

Comment:
With python's unitest framework anything you do to 'self' only effects the current test.  So no need to restore anything.   
For example see above how `self.emcc_args.append` is called but never inverted.

Reply:
Hmm, is that true even with other decorators applied? Some of them end up calling `f(self)` more than once. Seems like with the right order, this could end up persisting from one to the other?